source/ptreferential/ptreferential_utils.cpp

```diff
@@ -221,27 +221,21 @@ filter_on_period(const Indexes& indexes,
     if (since && until && until < since) {
         throw ptref_error("invalid filtering period");
     }
-    auto start = bt::ptime(data.meta->production_date.begin());
-    auto end = bt::ptime(data.meta->production_date.end());
+    auto start = bt::ptime(bt::neg_infin);
+    auto end = bt::ptime(bt::pos_infin);
+    if (requested_type == nt::Type_e::VehicleJourney) {
```

Comment:
Isn't that block useless?

Reply:
In case of vehicleJourney, the production period is used for the validity.

Reply:
The implementation for vehicle_journeys loop on the date between since and until.

Reply:
Maybe it's clearer to do that in the implementation of vehicle_journeys.